lte/gateway/python/magma/pipelined/ifaces.py

```diff
@@ -31,8 +32,33 @@ def monitor_ifaces(iface_names):
         yield from asyncio.sleep(POLL_INTERVAL_SECONDS)
 
 
-def get_mac_address(interface_name: str) -> str:
-    if_addresses = netifaces.ifaddresses(interface_name)[netifaces.AF_LINK]
-    if not if_addresses or not if_addresses[0].get('addr'):
-        raise ValueError(f"No mac address found for interface {interface_name}")
-    return if_addresses[0]['addr']
+def get_mac_address(
+        interface: Optional[str] = None,
+        ip4: Optional[str] = None,
+        ip6: Optional[str] = None,
```

Comment:
Why not just make it two or three methods?

Reply:
Done.